tools/utils/download_blender.sh

```diff
@@ -33,6 +36,7 @@ declare -A BLENDER_DOWNLOAD_URL_LINUX=(
     ["v2.81"]="https://download.blender.org/release/Blender2.81/blender-2.81a-linux-glibc217-x86_64.tar.bz2"
     ["v2.82"]="https://download.blender.org/release/Blender2.82/blender-2.82a-linux64.tar.xz"
     ["v2.83"]="https://download.blender.org/release/Blender2.83/blender-2.83.3-linux64.tar.xz"
+    ["v2.90"]="https://github.com/grische/blender/releases/download/v2.90.0-9de5adc6a1a9/blender-2.90.0-9de5adc6a1a9-linux64.tar.xz"
```

Comment:
I heard you try to get latest 2.90 beta URL from blender.org by using curl command or something else.
How about its progress?

Reply:
That is a different topic and is part of a complex CI pipeline here: https://github.com/grische/blender/blob/automatic_sync/.github/workflows/blender_create_release.yaml

The main problem why I created this huge pipeline linked above: 
Blender Beta and Alpha downloads are only available for ~24 hours. After that, the files are being deleted and replaced with new builds.

Apart from that the fact, that it is annoying to deal with, it has some impact on a CI pipeline:
If you dynamically fetch Blender downloads, you lose all stability guarantees.
1. As there are no checksums, you can never tell if you fetched the expected and tested binary or not (e.g. corrupted downloads)
1. You never know if you can create a PyPi release: Even if a master build succeeds today, it might fail tomorrow because the underlying download changed (e.g. the build of 2.90 yesterday and 2.90 two days ago have different python doc fixes applied)
1. You never know if a PR actually works: Similarly, a PR that had a successful CI run a week ago and got merged only today might fail, because the underlying download changed.
1. Tests are not reliable anymore: Even if nothing changed on the blender binary, you still need to keep the blender binary and the Cycles addon in sync. If you then sync the Cycles addon depending on the download, you could, for example, have different/unexpected pylint errors even though fake-bpy-module has not changed.

Only a manual and controlled update of the Blender download (i.e. via a PR: a build + tests) works in the context of a CI pipeline.

All in all: I did not create the Github Blender downloads for fun 😉

Reply:
Thanks for sharing the downside of keeping the latest build.
Taking account of these downside, it's reasonable to stick on the specific version.